llarp/router/abstractrouter.hpp

```diff
@@ -131,6 +131,36 @@ namespace llarp
     virtual void
     ModifyOurRC(std::function<std::optional<RouterContact>(RouterContact)> modify) = 0;
 
+    /// mark that think we will be invalid in the future, or nullopt to reset this to normal
+    inline void
+    InvalidAt(std::optional<TimePoint_t> maybe_future_time)
+    {
+      if (maybe_future_time)
+      {
+        // when we think we are invalid in the future we will publish slightly faster so we can keep
+        // the information as accurate as possible as the backend's guess ebbs and flows
```

Comment:
This faster publishing feels a bit like solving the issue the wrong way around: the blockchain time may be a bit inaccurate, but we should just have the client code be aware of that and simply stop using a router a couple hours before its unlock timestamp estimate so that the accuracy won't matter.

Reply:
fair.